tools/cfn2ts/lib/codegen.ts

```diff
@@ -385,6 +390,44 @@ export default class CodeGenerator {
     this.code.closeBlock();
   }
 
+  /**
+   * Emit a validator for the given construct
+   *
+   * The generated code looks like this:
+   *
+   * ```
+   * protected validate(): string[] {
+   *   const errors = new Array<string>();
+   *   // Validation here...
+   *   return errors;
+   * }
+   * ```
+   */
+  private emitConstructValidator(resourceType: genspec.CodeName) {
+    this.code.openBlock('protected validate(): string[]');
```

Comment:
Assume you're trying to use the validate API on constructs. This is going away with v2 - https://github.com/aws/aws-cdk-rfcs/blob/master/text/0192-remove-constructs-compat.md#08-validation-the-validate-hook-is-now-nodeaddvalidation

Use `addValidation()` instead.

Reply:
```
@aws-cdk/core: [96mlib/cloudformation.generated.ts[0m:[93m883[0m:[93m23[0m - [91merror[0m[90m TS2339: [0mProperty 'addValidation' does not exist on type 'ConstructNode'.
```

Reply:
cc @eladb

Reply:
oh wait. Is this because this doesn't exist on `constructs` pre-10.x?

Reply:
It does exist: https://github.com/aws/constructs/blob/117d3f1fee039eeac2d2aefa1b3e2d81530a63ff/src/construct.ts#L415